sweagent/frontend/src/Run.js

```diff
@@ -154,8 +157,44 @@ function Run() {
     }
   };
 
+  const checkScrollPosition = (ref, scrollStateRef, offset = 0) => {
+    scrollStateRef.current =
+      ref.current.scrollTop + ref.current.clientHeight + offset <
+      ref.current.scrollHeight;
+  };
+
+  const scrollToBottom = (ref, scrollStateRef) => {
+    if (!scrollStateRef.current) {
+      ref.current.scrollTop = ref.current.scrollHeight;
+    }
+  };
+
+  const scrollDetected = () => checkScrollPosition(logsRef, isLogScrolled, 58);
+  const scrollLog = () => scrollToBottom(logsRef, isLogScrolled);
+
+  const scrollDetectedEnv = () =>
+    checkScrollPosition(envFeedRef, isEnvScrolled);
+  const scrollEnv = () => scrollToBottom(envFeedRef, isEnvScrolled);
+
+  const scrollDetectedAgent = () =>
+    checkScrollPosition(agentFeedRef, isAgentScrolled);
+  const scrollAgent = () => scrollToBottom(agentFeedRef, isAgentScrolled);
+
   // Use effect to listen to socket updates
   React.useEffect(() => {
+    logsRef.current.removeEventListener("scroll", scrollDetected);
```

Comment:
why are we first removing and then adding the same listener?

Reply:
Actually, why do we need the listeners at all? Can't we just put this logic directly in the scroll functions?

Reply:
Removed the extraenous remote listeners commands. I believe we need the event listeners in order to continually update the state of whether the user has scrolled. Without it, I tried inserting the logic into the scroll functions, but had difficultly updating whether the user had scrolled or not depending on large message updates that would require explicitly saving past scroll positions. Please let me know if there is another solution or if you would recommend I continue to pursue this method.

Reply:
Sounds good to me!